compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1453,7 +1461,23 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         // frequently opened issues show.
         let opaque_ty_span = self.mark_span_with_reason(DesugaringKind::OpaqueTy, span, None);
 
-        let opaque_ty_def_id = self.local_def_id(opaque_ty_node_id);
+        let opaque_ty_def_id = match origin {
+            hir::OpaqueTyOrigin::TyAlias(def_id) => {
+                if let Some(def_id) = self.opt_local_def_id(opaque_ty_node_id) {
```

Comment:
I what case would this ever be `Some`?

Reply:
The 3 cases are coming from ...

```
stack backtrace:
   0: rust_begin_unwind
             at ./library/std/src/panicking.rs:556:5
   1: core::panicking::panic_fmt
             at ./library/core/src/panicking.rs:142:14
   2: <rustc_ast_lowering::LoweringContext>::lower_opaque_impl_trait
   3: <rustc_ast_lowering::LoweringContext>::lower_ty_direct
   4: <rustc_ast_lowering::LoweringContext>::lower_ty
             at ./compiler/rustc_ast_lowering/src/lib.rs:1197:26
   5: <rustc_ast_lowering::LoweringContext>::lower_assoc_ty_constraint::{closure#0}
             at ./compiler/rustc_ast_lowering/src/lib.rs:1079:34
   6: <rustc_ast_lowering::LoweringContext>::with_dyn_type_scope::<rustc_hir::hir::TypeBindingKind, <rustc_ast_lowering::LoweringContext>::lower_assoc_ty_constraint::{closure#0}>
             at ./compiler/rustc_ast_lowering/src/lib.rs:868:22
   7: <rustc_ast_lowering::LoweringContext>::lower_assoc_ty_constraint
             at ./compiler/rustc_ast_lowering/src/lib.rs:1071:21
   8: <rustc_ast_lowering::LoweringContext>::lower_angle_bracketed_parameter_data::{closure#2}
             at ./compiler/rustc_ast_lowering/src/path.rs:335:54
   9: <&mut <rustc_ast_lowering::LoweringContext>::lower_angle_bracketed_parameter_data::{closure#2} as core::ops::function::FnMut<(&rustc_ast::ast::AngleBracketedArg,)>>::call_mut
```

Reply:
Ok, this is fixed now. Have seen that we are creating a def id for the impl trait in lower_assoc_ty_contraint